### PR TITLE
Istanbul holiman

### PR DESCRIPTION
**Pull request by shemnon**
Adaptations of @holiman's istanbul tests as state test fillers.

**TODO:**
refill with latest solidity

### CODE REVIEW

**Comment by shemnon** [comment]
@holiman's tests these were derived from: https://github.com/holiman/IstanbulTests/tree/3c34f2c1a24175f5f4424b64b5e7ed6d822dc6af/GeneralStateTests

**Comment by holiman** [comment]
Good job, hats off!

File: src/GeneralStateTestsFiller/stChainId/chainIdGasCostFiller.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+              }
+            }
+          }
+        }
+      ],
+      "pre": {
+        "1000000000000000000000000000000000000000": {
+          "balance": "",
+          "//code": "record the gas, run the operation, then record the gas again.  Drop output, subtract 2 for GAS, what is left is the cost of the operation, store in storage 0x01.",
+          "//codeLLL": "(asm GAS CHAINID GAS SWAP1 POP SWAP1 SUB 5 SWAP1 SUB 0x01 SSTORE)",
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stChainId/chainIdGasCostFiller.json]: substract 2  or 5 ? 
looks a good asm code template for gasCost checkup for each opcode

File: src/GeneralStateTestsFiller/stSLoadTest/sloadGasCostFiller.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{
+    "sloadGasCost": {
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stSLoadTest/sloadGasCostFiller.json:2]: this test has a very good post condition!

File: src/GeneralStateTestsFiller/stSelfBalance/selfBalanceEqualsBalanceFiller.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                "0x01": "0x01"
+              }
+            }
+          }
+        }
+      ],
+      "pre": {
+        "1000000000000000000000000000000000000000": {
+          "balance": "0x01f4",
+          "//code": "{ [[ 1 ]] (EQ (BALANCE (ADDRESS)) (SELFBALANCE)) }",
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stSelfBalance/selfBalanceEqualsBalanceFiller.json]: really need to add lllc support for new opcodes

File: src/GeneralStateTestsFiller/stSelfBalance/selfBalanceFiller.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                "0x01": "0x01f4"
+              }
+            }
+          }
+        }
+      ],
+      "pre": {
+        "1000000000000000000000000000000000000000": {
+          "balance": "0x01f4",
+          "//code": "{ [[ 1 ]] (SELFBALANCE) }",
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stSelfBalance/selfBalanceFiller.json]: have some more ideas:

```
Test selfbalance inside codecall delegatecall staticcall ordinary calls
Test selfbalance in create opcode in inicode
Test selfbalance in tr init code
Suicide then test selfbalance return value
Selfbalance after revert operations. And after oog operations in a subcall that doesn't break the whole transaction execution
Selfbalance on different calldepth levels
* this are also to be added to a template for any kind of opcode

```

**Comment by winsvega** [comment]
This are very good tests, @shemnon. thanks! 
would look into the update for lllc so you don't  have to put the bytecode all the time

**Comment by winsvega** [comment]
latest solidity works again. try out this PR 
https://github.com/ethereum/solidity/pull/7327

it support CHAINID and SELFBALANCE

**Comment by shemnon** [comment]
Updated.  Old raw bytecode removed from fillers.

**Review by winsvega — APPROVED** [review]

**Comment by winsvega** [comment]
geth passes this tests

**Pull request merged**